unit-tests/src/org/commcare/android/tests/processing/KeyRecordTest.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * 
+ */
+package org.commcare.android.tests.processing;
+
+import static junit.framework.Assert.assertEquals;
+
+import org.commcare.android.database.user.models.ACase;
+import org.commcare.android.junit.CommCareTestRunner;
+import org.commcare.android.shadows.SQLiteDatabaseNative;
+import org.commcare.android.util.LivePrototypeFactory;
+import org.commcare.android.util.TestUtils;
+import org.commcare.cases.model.Case;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.annotation.Config;
+
+/**
+ * @author ctsims
+ *
+ */
+@Config(shadows={SQLiteDatabaseNative.class}, emulateSdk = 18, application=org.commcare.dalvik.application.CommCareApplication.class)
+@RunWith(CommCareTestRunner.class)
+public class KeyRecordTest {
+
+    //TODO: Move this to the application or somewhere better static
+    static LivePrototypeFactory factory = new LivePrototypeFactory();
+
+    @Before
+    public void setupTests() {
+        TestUtils.initializeStaticTestStorage();
+    }
+    
+    @Test
+    public void testKeyRecordParse() {
```

Comment:
A comment detailing what functionality this is testing would be great. What are keys? When does this happen in the app's logic? Do we expect to be in a particular state before or after a key parse? I know I could go read the code and figure all this stuff out, but a few lines of text here would be a much easier way of giving me that context.